pkg/services/accesscontrol/resourcepermissions/service.go

```diff
@@ -293,7 +293,9 @@ func (s *Service) SetPermissions(
 	ctx, span := tracer.Start(ctx, "accesscontrol.resourcepermissions.SetPermissions")
 	defer span.End()
 
-	if err := s.validateResource(ctx, orgID, resourceID); err != nil {
+	// permissions have not yet been set - validate the resource with a service account
+	identityCtx, _ := identity.WithServiceIdentitiy(ctx, orgID)
```

Comment:
Should this be done in [validateResource](https://github.com/grafana/grafana/pull/99709/files#diff-df31fc307ddb09b17b286316a8cc9b4a866578297cbc60f090ec1c3a065749a5R374)? 

I am not sure what's the intent with the validateResource ? Is it to check that the resource is valid ? or that the user is allowed to fetch the resource ?